### ISSUE TITLE
Repeated appearances of the `exceptional` condition

### ISSUE THREAD

**Issue by SiriosDev** [Pirate-Weather/pirate-weather-ha#276]
### Describe the bug

In recent days I have been experiencing repeated appearances of the `exceptional` state. Checking on merrysky it does not turn out to be anything particularly out of the ordinary

### Expected behavior

See an analogue to the conditions shown by other PW-based 

### Actual behavior

e.g. exactly now it is shown exceptional when the condition is trivially partly cloudy
(`eccezionale` mean `exceptional`)
![image](https://github.com/user-attachments/assets/84521587-01d7-4d38-bd58-0d2543bd0ab5)
![image](https://github.com/user-attachments/assets/b25bc22a-c593-4fe2-bbba-cb5287933f09)

### Home Assistant version

2024.7.4

### Integration version

1.5.2

### Other details

Unfortunately, the `exceptional` condition is poorly documented but browsing the core code [these](https://github.com/home-assistant/core/blob/537d7728a726816220be0ca7c7f68ecea7d835d1/homeassistant/components/nws/const.py#L34) seem to be the conditions that fall under the term `exceptional`

> ```py
> #...
> CONDITION_CLASSES: dict[str, list[str]] = {
>     ATTR_CONDITION_EXCEPTIONAL: [
>         "Tornado",
>         "Hurricane conditions",
>         "Tropical storm conditions",
>         "Dust",
>         "Smoke",
>         "Haze",
>         "Hot",
>         "Cold",
>     ],
> #...
> ```

### Troubleshooting steps

- [X] I have updated my Home Assistant installation to the latest version.
- [X] I have updated the Pirate Weather Integration to the latest version.
- [X] I have gone through the [documentation](https://pirateweather.net/en/latest/ha/) before opening this issue.
- [X] I have searched this repository and [API Repository](https://github.com/alexander0042/pirate-weather-ha) to see if the issue has already been reported.
- [X] I have restarted my Home Assistant installation.
- [X] I have queried the API in my browser to confirm the issue is not with the API.
- [X] I have written an informative title.

**Comment by cloneofghosts** [comment]
Thanks for opening this issue. Taking a look at the code its setup to return the `exceptional` condition if the condition code is `tornado` or `none`.

https://github.com/Pirate-Weather/pirate-weather-ha/blob/b09d1ffef04074981d96f0775fd5384324fafaea/custom_components/pirateweather/weather.py#L91-L92

Out of the two conditions the only one which can sometimes be returned by the API is `none`. The bigger question is why is `none` being returned as a condition.

Whereabouts are you approximately located so @alexander0042 can take a look and see what's going on. It might be related to the weird issues that people have been seeing with the API (see https://github.com/Pirate-Weather/pirateweather/issues/229). API version 2.1 should fix these issues but its only been released on the dev endpoint.

**Comment by SiriosDev** [comment]
ye ik the issue 229  of the main repo, the author looks a LOT like me after all 🤣. Joking aside, Could be the API, I didn't get around to taking the screenshot but indeed, now that I opened it even from the Breezy Weather app with PW as source, it showed None (before it updated). I was misled by merrysky 😅

**Comment by cloneofghosts** [comment]
> ye ik the issue 229 of the main repo, the author looks a LOT like me after all 🤣.

Yes, I realized after I posted that you had created that other issue.

> Could be the API, I didn't get around to taking the screenshot but indeed, now that I opened it even from the Breezy Weather app with PW as source, it showed None (before it updated). I was misled by merrysky 😅

Yes, it's an issue with the API returning `none` instead of the current condition which shouldn't be happening. I'd like to test to see if this issue is fixed in API V2.1 but I haven't been able to find a location to test with. I'll see if I can find one to test with or if you have a location to use feel free to share and I can see if V2.1 will fix this.

**Comment by SiriosDev** [comment]
I will do some tests, as soon as I find a new none/exceptional I will write in the coordinates, although perhaps (It is clear that it is an API-side problem) it would be appropriate to close this issue to continue on the original issue

**Comment by SiriosDev** [comment]
I will do some tests, as soon as I find a new none/exceptional I will write in the coordinates, although perhaps (It is clear that it is an API-side ) it would be appropriate to close this issue to continue on the original issue
